### ISSUE TITLE
Farm: auto-harvesting happens too often now, and is too high priority.

### ISSUE THREAD

**Issue by com1clyf3** [CompletelyFairGames/dwarfcorp#649]
When blocks are highlighted green to either be harvested or planted, player is able to cancel farming task. However, any other time during autofarming, the tiles cannot be right-clicked and autofarming canceled. Instead the info message "There's something in the way." will display.

![image](https://user-images.githubusercontent.com/6562812/38835714-ea0f0b90-4199-11e8-812a-46b1b4cfbc3f.png)

Dwarves can spend too much of their time farming, instead of other tasks, unless you throttle down your farm tasks after your initial farming in early game, so de-selecting farm tiles is one fix/player solution.

The other fix is more complicated and goes something like:

1. Measure the amount of resources the dwarf colony probably needs. For instance, if there are five dwarves in the colony, maybe they need at least fifty apples.  Perhaps this could be player decided, like the drop down menu for cooking.

2. Once this ordered limit is reached, the dwarves lower the priority of farming to pretty low.

3. If the resource dips under the limit, the task is then elevated back to high, since food = not dead.

9e581f9d

**Comment by mklingen** [comment]
Hmm okay I think the right click is incorrectly checking to see if there's something growing already before allowing you to cancel.

As for the balance, we could also just make plants grow more slowly/randomly so that all of the field isn't up for harvest at the same time.

**Comment by com1clyf3** [comment]
Let's just do the fix first.

I think it's just best to notice the odd balance design for now. We'll polish it later when we have a better vision on what player experience is really supposed to be.

**Comment by mklingen** [comment]
@com1clyf3 , @Blecki  is this still an issue?

**Comment by Blecki** [comment]
Yep. The cancel tool is the solution though it doesn't actually work yet.

**Comment by mklingen** [comment]
Can confirm that cancel tool now works for auto harvest. Leaving this open for balance issues.
